api/src/test/java/org/openmrs/module/idgen/service/IdentifierSourceServiceTest.java

```diff
@@ -62,6 +63,13 @@ public void generateIdentifiers_shouldReturnIdentifiersOfCorrectSize() throws Ex
 		List<String>  sig = identifierSourceService.generateIdentifiers(is, 7, "hello");
 		Assert.assertEquals(sig.toString(), "[G-0, H-8, I-5, J-3, K-1, L-9, M-7]");
 	}
+	
+	@Test
+	public void generateIdentifiers_shouldReturnLocationPrefixedIdentifiers() {
+		IdentifierSource is = identifierSourceService.getIdentifierSource(8);
+		List<String>  sig = identifierSourceService.generateIdentifiers(is, 4, null);
+		Assert.assertEquals(sig.toString(), "[AFDEL-000005, AFDEL-000006, AFDEL-000007, AFDEL-000008]");

```

Comment:
Couldn't there be a way to show through this test that the prefixes are dynamic? Please also use generic names for the codes (no more references to delegations and sub-delegations).